doc/aaa/radius_authentication.md

```diff
@@ -0,0 +1,634 @@
+for AAA, but not for nameservice. If another nameservice like LDAP is
+not used, a RADIUS NSS plugin needs to be used.
+
+The nss_radius plugin will load the configuration file /etc/radius_nss.conf.
+It will also use the Management-Privilege-Level (MPL)  attribute (type 136)
+information obtained from the RADIUS server during authentication, to
+determine the user information that is needed to create a local user, at
+the time of first login. The information of this local user is returned
+during a getpwnam_r() library call.
+
+There is an option to map all users at a privilege level to one user as
+specified in the configuration file /etc/radius_nss.conf. This option is
+the many_to_one=y option. When this option is in effect, no local user is
+created at the time of login. Because of the mapping, not all features
+may apply, since the LOGNAME of the user, and the mapped uid may not match.
+
+For SSH login, Note 1. under sequence of events during RADIUS authenticated
+login will apply. The configuration option for this Note 1. to apply
+is many_to_one=a.
+
+
+### User Privilege Table
+
+By default, user information is returned as follows for users authenticated
+through RADIUS.
+
+| MPL (privilege)| user info      | uid  | gid  | secondary grps | shell     |
+| -------------- | -------------- | ---- | ---- | -------------- | --------- |
+| 15             | remote_user_su | 1000 | 1000 | sudo,docker    | /bin/bash |
+| 1-14           | remote_user    |65534 |65534 | users          | /bin/rbash|
+
+
+A different mapping of MPL can be configured in the /etc/radius_nss.conf,
+on the switch. For example, if the following lines were added to the
+configuration.
+
+```
+user_priv=15;pw_info=remote_user_su;uid=1000;gid=1000;group=sudo,docker;dir=/home/admin;shell=/bin/bash
+user_priv=7;pw_info=netops;uid=2007;gid=100;group=users;dir=/home/netops;shell=/bin/rbash
+user_priv=1;pw_info=operator;uid=2001;gid=100;group=users;dir=/home/operator;shell=/bin/rbash
+```
+
+the mapping of MPL would be as follows.
+
+| MPL (privilege)| user info      | uid  | gid  | secondary grps | shell     |
+| -------------- | -------------- | ---- | ---- | -------------- | --------- |
+| 15             | remote_user_su | 1000 | 1000 | sudo,docker    | /bin/bash |
+| 14-7           | netops         | 2007 |  100 | users          | /bin/rbash|
+| 1-6            | operator       | 2001 |  100 | users          | /bin/rbash|
+
+
+
+## ConfigDB Schema
+
+### AAA Table Schema
+
+```
+; Key
+aaa_key              = "authentication"   ; AAA type
+; Attributes
+login                = LIST(1*32VCHAR)   ; pam modules for particular protocol, now only support login for (local, tacacs+, radius). Legal combinations are [ [local], [local, tacacs+ ], [local, radius], [tacacs+, local], [radius, local] ]
+failthrough          = "True" / "False"  ; failthrough mechanism for pam modules
+debug                = "True" / "False"  ; debug logs for nss or pam modules
```

Comment:
Which log file?

How is this flag going to behave, for tacacs & local auth ?

Reply:
Most PAM modules have a debug option which sends messages to syslog(3). This has a direct mapping to that option

```
OPTIONS
       debug
           Turns on debugging via syslog(3).
```

Reply:
Thanks. So we are going to benefit from this flag, for both Tacacs & Radius. Pretty good.